### ISSUE TITLE
[0.F planning] Auditing gun, magazine, and holster volumes

### ISSUE THREAD

**Issue by LaVeyanFiend** [CleverRaven/Cataclysm-DDA#37374]
**Is your feature request related to a problem? Please describe.**  
Guns and magazines are mostly still defined volumetrically by 250ml*n for both their own volume and the magwell's, in the case of the former. In general:
-Pistols & magazines are way too big
-Longguns are too small

**Describe the solution you'd like**  
Running all guns & magazines through ImageJ(#33711, #34666, and #36838 used it) to get their volumes as gun-shaped prisms, then updating spare ammo containers & holsters on some basis of them.

**Describe alternatives you've considered**  

**Additional context**  
"Shoving a rifle in cargo pants" will be a dead meme, even before the nested containers update.

**Comment by GGgatherer** [comment]
To really make this work you'd need to add "longest dimension" property to items and containers (in addition to volume). Then some rifle may still take only 2L by volume but you wouldn't be able to store it even in a 12L(or whatever) backpack because it's longer that the backpack.

**Comment by LaVeyanFiend** [comment]
I've actually already talked with KorG about that, longest dimension will be added to deal with the holster refactor.

**Comment by KorGgenT** [comment]
yeah it'll happen during the nested containers project and holster refactor at some point

**Comment by LaVeyanFiend** [comment]
Posting a tutorial so people other than tenmillimaster and I can help:

0. Download [the free program ImageJ](https://imagej.nih.gov/ij/).

Tutorial for gunz, using OA-93 pistol:
![image](https://user-images.githubusercontent.com/51099123/73503421-361b3a00-439a-11ea-8c85-2681a4d80cba.png)


1. Get image of gun, with a known length and width.
2. Make the gun solid black on white background by using this option:
![image](https://user-images.githubusercontent.com/51099123/73503155-7332fc80-4399-11ea-9a90-aae164b8735a.png)
3. Fill gaps in gun with Fill Holes.
![image](https://user-images.githubusercontent.com/51099123/73503174-8219af00-4399-11ea-9544-be28598819e0.png)
4.  Click the gun with the wand tool active.
![image](https://user-images.githubusercontent.com/51099123/73503200-978ed900-4399-11ea-8eae-2f112fa79d3a.png)
5.  Crop with ctrl+shift+x.
6. Set the scale's known pixels(ex. 1118 pixels from pistol math) to its known length of 17in.
![image](https://user-images.githubusercontent.com/51099123/73503273-d6249380-4399-11ea-8d2c-d4e82ea9910d.png)
![image](https://user-images.githubusercontent.com/51099123/73503286-e50b4600-4399-11ea-95ef-60b794ba6d2d.png)
7. Analyze particles to get surface area of the gun.
![image](https://user-images.githubusercontent.com/51099123/73503461-521edb80-439a-11ea-9e61-e36aff515c79.png)
![image](https://user-images.githubusercontent.com/51099123/73503482-61058e00-439a-11ea-96ec-0411c7912052.png)
8. Multiply by gun's width, and convert to ml(if necessary).

**Comment by LaVeyanFiend** [comment]
I'll be able to put in a lot more work next week, once spring break starts. Begone stalebot!

**Comment by LaVeyanFiend** [comment]
Can stale bot be turned off for this issue?
(bump)

**Comment by LaVeyanFiend** [comment]
Updated tutorial for adding longest_side in light of #40186.
